### PR TITLE
Mr robot

### PR DESCRIPTION
**Pull request by NOEMOJI041**
MR ROBOT WIKI
ADDED :-
1) OVERVIEW
2) CODE EXPLANATION
3) MISCELLANEOUS

### CODE REVIEW

File: markdown/projects/mr-robot/major-code/arduino.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      attachInterrupt(digitalPinToInterrupt(encoderLPin1), updateEncoder_L, CHANGE);
+      attachInterrupt(digitalPinToInterrupt(encoderLPin2), updateEncoder_L, CHANGE);
+    
+      attachInterrupt(digitalPinToInterrupt(encoderRPin1), updateEncoder_R, CHANGE);
+      attachInterrupt(digitalPinToInterrupt(encoderRPin2), updateEncoder_R, CHANGE);
+
+
+    }
+
+This code snippet is used to set up the various peripherals and settings needed for the robot to operate, including the ROS node, PWM signals for the motors, encoder signals for the wheels, and interrupt service routines for updating the encoder counts.
```
Comment (Kartik9250) [inline markdown/projects/mr-robot/major-code/arduino.rst:64]: Peripherals is not the right word to be used to address the components/parts here

File: markdown/projects/mr-robot/major-code/arduino.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+.. code:: C
+
+    void LpwmCb(const std_msgs::Int32& pwm){
+
+        if(pwm.data == 0) { ledcWrite(ledChannel1, 0); ledcWrite(ledChannel2, 0); }
+        if(pwm.data > 0)  { ledcWrite(ledChannel1, abs(pwm.data)); ledcWrite(ledChannel2, 0); }
+        if(pwm.data < 0)  { ledcWrite(ledChannel1, 0); ledcWrite(ledChannel2, abs(pwm.data));}
+    }
+
+This code defines two callback functions, LpwmCb and RpwmCb, which take a std_msgs::Int32 object as input. The functions control the speed of four motors by writing PWM signals using the ledcWrite function.
```
Comment (Kartik9250) [inline markdown/projects/mr-robot/major-code/arduino.rst:13]: Four motors were not used. Write two instead.

Comment (Kartik9250) [reply markdown/projects/mr-robot/major-code/arduino.rst:13]: Add Inline code for highlighting the different code components: LpwmCb, std_msgs

File: markdown/projects/mr-robot/major-code/arduino.rst
Comment (Kartik9250) [inline markdown/projects/mr-robot/major-code/arduino.rst:1]: Define how arduino is communicating the Ros master node

File: markdown/projects/mr-robot/major-code/arduino.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+      right_enc_pub.publish(&encoder_msg_right);
+
+        encoder_msg_left_error.data = rpmL_error;
+      left_enc_error.publish(&encoder_msg_left_error);
+
+      encoder_msg_right_error.data = rpmR_error;
+      right_enc_error.publish(&encoder_msg_right_error);
+    }
+
+The function publish_encoder_data() is used to publish the encoder values and any associated errors to their respective ROS topics.
```
Comment (Kartik9250) [inline markdown/projects/mr-robot/major-code/arduino.rst:92]: inline coments

**Review by Kartik9250 — CHANGES_REQUESTED** [review]
Use the following tag in order to make it resourceful & easier to read:

- Inline code tags
- Notes, Warnings, Seealso
- Link and cross reference our documentation pages.
- Use the Additional images that were added.
- Proof read the documentation and verify the provided information.
- Add references for the various websites for detailed information.

**Review by namikxgithub — APPROVED** [review]

**Comment by jasmeet0915** [comment]
Some changes are required @NOEMOJI041 :
cc: @Kartik9250 @namikxgithub 
 - Mention MR Robot fullform in Project intro(overview) page and it should be written as MR Robot and not Mr. Robot since it is an Acronym.
- Add image captions 
 - Add official Links for packages and hardware components in Working section
 - Add tf tree for the robot in working section
 - Include ros computation graph of nodes in launching section to show topics and nodes that are launched. This can be done using [rqt_graph](http://wiki.ros.org/rqt_graph) node.
 - Doesn't need such big description for Master-Slave. We already cover this in ROS Intro. This section should be named different and it should contain a block diagram showing where the different nodes of the setup run physically. This block diagram is a different from ROS computation graph and not necessarily needs to include topics. But should show where different nodes run (pi, laptop/pc etc.)

**Comment by NOEMOJI041** [comment]
Fixed:-
1)Mr. Robot ------> MR Robot
2)Information Check
3)Deleted Extra from Master-Slave and changed its name
Added:-
1)inline code tags
2)Image Captions
3)Official Links in working.rst
4)Note is added in esp-diff-tf.rst and Warning is added to bash.rst
Note:-
tf tree and rqt_graph will be added according to the latest test.

**Comment by NOEMOJI041** [comment]
Added TF TREE and Node Graph

File: markdown/projects/mr-robot/mr-index.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
+MR-ROBOT
+========
+
+
+Mr robot is autonomous navigation robot made by A.T.O.M Robotics capable of doing multiple day to day operations such as mapping, navigation for transportation, sanitaion etc. Various other operations can also be performed thanks to its modularity.
```
Comment (Kartik9250) [inline markdown/projects/mr-robot/mr-index.rst]: Mr robot is an* autonomous

Comment (NOEMOJI041) [reply markdown/projects/mr-robot/mr-index.rst]: done

**Review by Kartik9250 — APPROVED** [review]
Minor changes required, would be updated in next ver

**Pull request merged**